lib/scholar/shared.ex

```diff
@@ -66,4 +66,27 @@ defmodule Scholar.Shared do
               "invalid value for :weights option: expected list or tensor of positive numbers of size #{num_samples}, got: #{inspect(weights)}"
     end
   end
+
+  deftransform valid_broadcast?(n_dims, shape1, shape2) do
```

Comment:
I think the function should be called `validate_broadcast!`, as its main goal is to raise if something is wrong, rather than question mark.